data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/DecryptionKeyGenerator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.bls.keystore;
+
+import java.util.Objects;
+import org.apache.tuweni.bytes.Bytes;
+import org.bouncycastle.crypto.Digest;
+import org.bouncycastle.crypto.generators.PKCS5S2ParametersGenerator;
+import org.bouncycastle.crypto.generators.SCrypt;
+import org.bouncycastle.crypto.params.KeyParameter;
+import org.bouncycastle.crypto.util.DigestFactory;
+import tech.pegasys.artemis.bls.keystore.model.KdfParam;
+import tech.pegasys.artemis.bls.keystore.model.Pbkdf2Param;
+import tech.pegasys.artemis.bls.keystore.model.SCryptParam;
+
+public class DecryptionKeyGenerator {
+  public static Bytes generate(final byte[] password, final KdfParam kdfParam) {
+    Objects.requireNonNull(password);
+    Objects.requireNonNull(kdfParam);
```

Comment:
updated as advised, however, instead of using guava, I have introduced custom helper class which raises our custom exception (but using exact same method signatures).